sub_cal/sub_cal/src/main/java/model/Operation.java

```diff
@@ -0,0 +1,72 @@
+package model;
+
+import java.math.BigDecimal;
+import java.util.Stack;
+import java.util.StringTokenizer;
+public class Operation {
+    private Stack<Double> numberStack = new Stack<>();
+    private Stack<String> operStack = new Stack<>();
+    private static final int CONVERTTONEGATIVE = -1;
+
+    public String operate(String expression) {
+
+        StringTokenizer st = new StringTokenizer(expression);
+
+        classifyOperatorAndNumber(st);
+
+        return calculate();
+    }
+    private void classifyOperatorAndNumber(StringTokenizer st){
+
+        while(st.hasMoreTokens()){
+            String word = st.nextToken();
+            Operator operator = null;
+            if(!(Operator.getOperator(word).isEmpty())){
+                operator = Operator.getOperator(word).get();
+            }
+
+            switch (operator != null ? operator : operator.NULL){
+                case MULTIPLY : case DIVIDE:
+                    numberStack.push(Double.parseDouble(st.nextToken()));
+                    calculateMultiplyOrDivide(operator.toString());
+                    break;
+                case PLUS : case MINUS:
+                    operStack.add(operator.toString());
+                    break;
+                case NULL:
+                    if(operStack.isEmpty() || operStack.pop().equals("PLUS")){
+                        numberStack.push(Double.parseDouble(word));
+                        break;
+                    }
+
+                    numberStack.push(CONVERTTONEGATIVE*Double.parseDouble(word));
+                    break;
+            }
+        }
+
+
+    }
+
+    private String calculate(){
+        while(true){
+            if(numberStack.size() == 1) break;
```

Comment:
```suggestion
        while(numberStack.size() > 1){
```

`while(true)` 후 `break`보다는 적절한 조건을 넣어주는 건 어떨까요?

Reply:
while(true) 로 하는 게 습관이었는데 앞으로는 적절한 조건을 넣어주는 식으로 바꿔보겠습니다 !